hasktorch/src/Torch/Tensor.hs

```diff
@@ -30,7 +30,7 @@ import Torch.TensorOptions
 
 type ATenTensor = ForeignPtr ATen.Tensor
 
-data Tensor = Tensor ATenTensor
+newtype Tensor = Tensor ATenTensor
```

Comment:
I think we should rename this to `Unsafe` as well. No one should construct them by hand.

Reply:
Unsafe here would apply to the name of the constructor, while the type would remain Tensor, is that in line with what you're thinking?

Reply:
Yes, exactly.